db/migrate/20240817155611_split_public_timelines_setting.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class SplitPublicTimelinesSetting < ActiveRecord::Migration[7.1]
```

Comment:
I wasn't sure on the exact way to do this migration, but I've tested it locally a bunch and it appears to work.

Reply:
> The goal is to inherit the previous setting for both, if a previous setting existed, otherwise use the defaults (allow timeline_preview_local, prevent timeline_preview_remote).

The logic looks more/less correct to me ... I do wonder for perf reasons if a) should be "post migrate"? b) it might be gnarly with the settings hash storage, but could we this via direct sql w/out needing the AR class in the migration?

> Reverting the setting in the down is also a bit chaotic, and also kinda unneeded because, well, do we ever do down migrations post-merge of a migration?

If the data is truly destroyed and not recoverable/rebuildable, it should raise the irreversible migration error and not just approximate something. I'd say this case we can (at least if done immediately) in fact reconstruct the before state from the after state, so its fine to keep it.

Reply:
I've been having a think about this further, and I'm not sure we actually even want these routes, instead I think we'd want like `GET /api/v1/timelines/preview/local` or something. That way we avoid all this complicated boolean logic.